### PR TITLE
chore: get rid of k8s.io/kubernetes dep

### PR DESCRIPTION
**Pull request by makkes**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

The k8s.io/kubernetes dependency is problematic because of the many replaces it brings with it. This makes it hard to import external-snapshotter in 3rd party projects without having a lot of `replace` directives in those projects as well.

The only usage of that import was to get two very simple utility functions so I copied them over to this repo.

**Which issue(s) this PR fixes**:

no issue

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/utils/util.go [hunk 1]
```diff
@@ -109,6 +108,41 @@ var SnapshotterListSecretParams = secretParamsMap{
 	secretNamespaceKey: PrefixedSnapshotterListSecretNamespaceKey,
 }
 
+// ContainsString checks if a given slice of strings contains the provided string.
+// If a modifier func is provided, it is called with the slice item before the comparation.
+func ContainsString(slice []string, s string, modifier func(s string) string) bool {
```
Comment (yuxiangqian) [inline pkg/utils/util.go:113]: nit, can we remove the modifier? it appears its not used at all in this repo (i.e., all nils).

File: pkg/utils/util.go [hunk 2]
```diff
@@ -109,6 +108,41 @@ var SnapshotterListSecretParams = secretParamsMap{
...
+		if modifier != nil && modifier(item) == s {
+			return true
+		}
+	}
+	return false
+}
+
+// RemoveString returns a newly created []string that contains all items from slice that
+// are not equal to s and modifier(s) in case modifier func is provided.
+func RemoveString(slice []string, s string, modifier func(s string) string) []string {
```
Comment (yuxiangqian) [inline pkg/utils/util.go:127]: ditto

File: pkg/utils/util.go (hunk 1 above)
Comment (makkes) [reply pkg/utils/util.go]: done

File: pkg/utils/util.go (hunk 2 above)
Comment (makkes) [reply pkg/utils/util.go]: done

**Comment by makkes** [comment]
Thanks, @yuxiangqian for reviewing. I have addressed your comments.

**Comment by yuxiangqian** [comment]
lgtm,  will wait until all tests passed to approve.

**Comment by makkes** [comment]
@yuxiangqian would you mind having a look, please? 🙇‍♂️

**Pull request merged**